persist/sqlite/consensus.go

```diff
@@ -644,15 +826,23 @@ func (s *Store) LastCommittedIndex() (index types.ChainIndex, err error) {
 	return
 }
 
+// ResetLastIndex resets the last indexed tip to trigger a full rescan.
+func (s *Store) ResetLastIndex() error {
+	_, err := s.db.Exec(`UPDATE global_settings SET last_indexed_tip=$1`, encode(types.ChainIndex{}))
+	return err
+}
+
 func setLastCommittedIndex(tx *txn, index types.ChainIndex) error {
 	_, err := tx.Exec(`UPDATE global_settings SET last_indexed_tip=$1`, encode(index))
 	return err
 }
 
 func insertAddressStatement(tx *txn) (*stmt, error) {
+	// the on conflict is effectively a no-op, but enables us to return the id of the existing address
 	return tx.Prepare(`INSERT INTO sia_addresses (sia_address, siacoin_balance, immature_siacoin_balance, siafund_balance) VALUES ($1, $2, $2, $3) ON CONFLICT (sia_address) DO UPDATE SET sia_address=EXCLUDED.sia_address RETURNING id, siacoin_balance, immature_siacoin_balance, siafund_balance`)
 }
 
 func insertIndexStmt(tx *txn) (*stmt, error) {
+	// the on conflict is effectively a no-op, but enables us to return the id of the existing index
```

Comment:
Why can't we use `DO NOTHING RETURNING `id` here instead of performing a `no-op` `DO UPDATE SET`?

Reply:
`DO NOTHING` will not execute the `RETURNING` clause if there’s a conflict.